lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -138,8 +138,13 @@ public static ColumnReader create(PrimitiveField field, DateTimeZone timeZone, A
             if (REAL.equals(type) && primitiveType == FLOAT) {
                 return new FlatColumnReader<>(field, ValueDecoders::getRealDecoder, INT_ADAPTER, memoryContext);
             }
-            if (DOUBLE.equals(type) && primitiveType == PrimitiveTypeName.DOUBLE) {
-                return new FlatColumnReader<>(field, ValueDecoders::getDoubleDecoder, LONG_ADAPTER, memoryContext);
+            if (DOUBLE.equals(type)) {
+                if (primitiveType == PrimitiveTypeName.DOUBLE) {
+                    return new FlatColumnReader<>(field, ValueDecoders::getDoubleDecoder, LONG_ADAPTER, memoryContext);
+                }
+                if (primitiveType == FLOAT) {
+                    return new FlatColumnReader<>(field, TransformingValueDecoders::getFloatToDoubleDecoder, LONG_ADAPTER, memoryContext);
+                }
```

Comment:
Don't we need to add `throw unsupportedException(type, field);`?

Reply:
I'm leaving that unchanged for now, another PR https://github.com/trinodb/trino/pull/15713 is going to make this stricter.